kiali/kiali
# Iter8 fixes after UI end-to-end

More work to consolidate an end-to-end flow for Iter8 extension.

cc @jadeyliu939 
cc @kiali/iter8 

Frontend PR
https://github.com/kiali/kiali-ui/pull/1708

Main work on this PR:
- Avoid to pull Iter8 dependencies
- Create a static model of Iter8 entities to map the CRD. 
- Pros: avoid conflict with dependencies (Iter8 pulls different golang deps versions than Kiali and forces to non desired upgrades).
- Cons: this needs to be maintained on Iter8 changes and some logic (initialization of some default value) should be added.
- Refinements on endpoints and logic.

How to test in minikube:
- Install minikube + bookinfo.
- Install iter8 in minikube:
```
kubectl apply \
    -f https://raw.githubusercontent.com/iter8-tools/iter8-analytics/master/install/kubernetes/iter8-analytics.yaml \
    -f https://raw.githubusercontent.com/iter8-tools/iter8-controller/master/install/iter8-controller.yaml
```
- Enable iter8 extension in kiali config:
```
extensions:
  threescale:
    adapter_name: threescale
    adapter_port: 3333
    adapter_service: threescale-istio-adapter
    enabled: false
  iter8:
    enabled: true
```
- Manually:
-- Check Iter8 list page is enabled.
-- Check an experiment on reviews service can be created using reviews-v1 and reviews-v2.
-- Check that experiment is being updated with the progress.
-- Delete is not yet in implemented (I will remove do not merge when it's implemented).

Note, the goal of this PR (and UI) is to provide a Step 0 status of a very basic end-to-end CRUD.
